test/e2e/util/util.go

```diff
@@ -191,6 +191,28 @@ func GetHostFromClientConfig() (string, error) {
 	return clientCfg.Host, nil
 }
 
+func GetManagedCluster(dynamicClient dynamic.Interface, clusterName string) (*unstructured.Unstructured, error) {
+	cluster, err := dynamicClient.Resource(ManagedClusterGVR).Get(context.TODO(), clusterName, metav1.GetOptions{})
+	if err != nil {
```

Comment:
this is to allow  running e2e separately against ACM. 

I would want this to use clusterClient  instead of dynamicClient  and also other  cluster related func here as  a follow up @zhiweiyin318 @ldpliu 

/apprrove